### ISSUE TITLE
error: no matching function for call to ‘torch::class_<torch_ths::FasterTransformerEncoder>::class_(const char [25])’    torch::jit::class_<torch_ths::FasterTransformerEncoder>("FasterTransformerEncoder")

### ISSUE THREAD

**Issue by Hello-Pytorch** [NVIDIA/DeepLearningExamples#712]
Hello, i can succeed to compile custom c++ function, but fail to compile custom c++ class, the error is:
```
v3.0/fastertransformer/th_op/ft_ths_op.cc:27:85: error: no matching function for call to ‘torch::class_<torch_ths::FasterTransformerEncoder>::class_(const char [25])’
   torch::jit::class_<torch_ths::FasterTransformerEncoder>("FasterTransformerEncoder")
```
all class in ft_ths_op.cc can't be compiled.

My environment:
- Pytorch 1.6
- CUDA 10.2
- GCC 7.5

**Comment by byshiue** [comment]
Public PyTorch release has different C++ APIs compared with container images from NVIDIA.
Please replace the code in ft_ths_op.cc with “torch::jit::class_<torch_ths::FasterTransformerEncoder>("FasterTransformer", "Encoder")”
The importing code should be also modified from “torch.classes.FasterTransformerEncoder” to“torch.classes.FasterTransformer.Encoder”
Decoder and decoding are likewise.
We will fix this in FT 3.1.

**Comment by Hello-Pytorch** [comment]
> torch.classes.FasterTransformer.Encoder

Everything seems ok, thank you very much! Close this.
